Automattic/wp-calypso
# Components: Change CreditCardPage from component to block

In response to #7022 this PR refactors `<CreditCardPage />` and moves it to blocks folder with the more accurate name `<CreditCardForm />`. To help that happen I also decided to rename the existing component `<CreditCardForm />` to `<CreditCardFormFields />` which better explains it's purpose.

As part of this change I also added `<CreditCardPage />` example to devdocs and missing README file.

![screen shot 2016-08-25 at 17 02 07](https://cloud.githubusercontent.com/assets/699132/17996372/b1e8ab1c-6b68-11e6-9d22-47c06a4fab42.png)

### Testing
Test live: https://calypso.live/?branch=update/rename-credit-card-components

1. Go to DevDocs: https://calypso.live/devdocs/blocks.
2. You should see Credit Card form there. You can even play with it. Validation works as in other places, but the possibility to Save Card is disabled :) 
3. Go to Add Credit Card page: https://calypso.live/payment-methods/add-credit-card.
4. Credit Card form should look and work like before.
5. Go to Manage Purchases page (https://calypso.live/purchases).
6. Select a purchase paid with credit card.
7. Click on Edit Payment Method link.
8. Make sure Credit Card form looks like before.
9. Go through purchase flow and make sure Credit Card form looks like before.
